packages/api-server/src/error.ts

```diff
@@ -0,0 +1,160 @@
+//  Error code from JSON-RPC 2.0 spec
+//  reference: http://www.jsonrpc.org/specification#error_object
+
+import { JSONRPCError } from "jayson";
+import { COMPATIBLE_DOCS_URL } from "./methods/constant";
+
+export interface ErrorKind {
+  code: number;
+  message: string;
+}
+
+export class AppError extends Error implements JSONRPCError {
+  code: number;
+  data?: object;
+
+  constructor(kind: ErrorKind, data?: object) {
+    super(kind.message);
+    this.name = kind.message;
+    this.code = kind.code;
+    this.data = data;
+  }
+
+  addContext(context: Object): AppError {
+    this.data = {
+      ...this.data,
+      ...context,
+    };
+    return this;
+  }
+}
+
+const DEFAULT_ERROR_CODE: number = -32000;
+
+export const ERRORS = Object.freeze({
```

Comment:
```suggestion
export const ERRORS: Readonly<{ [key: string]: ErrorKind }> = Object.freeze({
```

Add type to `ERRORS`, and remove all `as ErrorKind`

Reply:
Good idea. I will have a try!